NDIndex item_pattern android support?

## NDIndex.java item_pattern android support?

Hello, I'm having the following problem with Android Studio, when I tried to apply transformations NDIndex have an error in the item_pattern:
`private static final Pattern ITEM_PATTERN =
            Pattern.compile("(\\*)|((-?\\d+|\\{})?:(-?\\d+|\\{})?(:(-?\\d+|\\{}))?)|(-?\\d+|\\{})");`

According to what I've searched so far android studio need the `\\` before each `}` `{`.

My question: Is there a way to work out this problem to make it work on android studio?

Thanks in advance :)

@frankfliu That's my problem, I'm not able to reproduce this line in Android Studio. I have this pipeline transformations:
`Pipeline pipeline = new Pipeline();
        pipeline.add(new CenterCrop(240, 240))
                .add(new ToTensor())
                .add(new Normalize(
                        new float[] {0.485f, 0.456f, 0.406f},
                        new float[] {0.229f, 0.224f, 0.225f}));`
And with `CenterCrop` I'm having the following issue:
`2020-10-22 00:19:09.232 27454-27454/com.example.demotienda E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.demotienda, PID: 27454
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.demotienda/com.example.demotienda.MainActivity}: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 16
    (\*)|((-?\d+|\{})?:(-?\d+|\{})?(:(-?\d+|\{}))?)|(-?\d+|\{})
                    ^
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2830)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2905)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1606)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:169)
        at android.app.ActivityThread.main(ActivityThread.java:6595)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 16
    (\*)|((-?\d+|\{})?:(-?\d+|\{})?(:(-?\d+|\{}))?)|(-?\d+|\{})
                    ^
        at java.util.regex.Pattern.compileImpl(Native Method)
        at java.util.regex.Pattern.compile(Pattern.java:1340)
        at java.util.regex.Pattern.<init>(Pattern.java:1324)
        at java.util.regex.Pattern.compile(Pattern.java:946)
        at com.example.demotienda.MainActivity.onCreate(MainActivity.java:62)
        at android.app.Activity.performCreate(Activity.java:7016)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2783)
        	... 9 more`

--- a/api/src/main/java/ai/djl/ndarray/index/NDIndex.java
+++ b/api/src/main/java/ai/djl/ndarray/index/NDIndex.java
@@ -46,8 +46,10 @@ import java.util.stream.Stream;
  */
 public class NDIndex {
 
+    /* Android regex requires escape } char as well */
     private static final Pattern ITEM_PATTERN =
-            Pattern.compile("(\\*)|((-?\\d+|\\{})?:(-?\\d+|\\{})?(:(-?\\d+|\\{}))?)|(-?\\d+|\\{})");
+            Pattern.compile(
+                    "(\\*)|((-?\\d+|\\{\\})?:(-?\\d+|\\{\\})?(:(-?\\d+|\\{\\}))?)|(-?\\d+|\\{\\})");
 
     private int rank;
     private List<NDIndexElement> indices;

